### DISCUSSION TITLE
Unable to deploy Teleport: `tls: failed to verify certificate: x509: certificate is valid for...`

### DISCUSSION THREAD

**Discussion by Dentrax** [gravitational/teleport#49813] [Q&A] [answered]
```
helm install teleport-cluster teleport/teleport-cluster \
  --set clusterName=teleport-cluster.teleport.svc.my-test-cluster \
  --set=operator.enabled=true \
  --set=global.clusterDomain=my-test-cluster \
  --set=persistence.storageClassName=local-hostpath
```

(`local-hostpath` is OpenEBS)

Results:

```
kubernetes failed to validate token: [invalid bearer token, token audiences [&#34;https://kubernetes.default.svc.my-test-cluster&#34;] is invalid for the target audiences [&#34;https://kubernetes.default.svc&#34; &#34;teleport-cluster.teleport.svc.my-test-cluster&#34;]], pinging proxy to determine signature algorithm suite
	Get &#34;https://teleport-cluster-auth.teleport.svc.my-test-cluster:3025/webapi/find&#34;: tls: failed to verify certificate: x509: certificate is valid for teleport-cluster-auth-56fb8b994b-g27hc, *.teleport.cluster.local, teleport.cluster.local, not teleport-cluster-auth.teleport.svc.my-test-cluster] service/connect.go:102
```

* I've checked the `ConfigMap`s and ensured those are correctly set
* Restarted all `Pod`s
* I have `my-test-cluster` in the `CoreDNS`, I able to resolve `teleport-cluster-auth.teleport.svc.my-test-cluster` but not `cluster.local`
* I see the [similar discussion here](https://github.com/gravitational/teleport/discussions/29963), but it didn't work for me, or its a bit different case

**Comment by webvictim** [comment]
@hugoShaka I think this might be the same issue as https://github.com/gravitational/teleport/issues/49756?

There is a PR to fix it: https://github.com/gravitational/teleport/pull/49796

**Reply by hugoShaka** [reply]
Yes, this is a 17.0.3 regression I introduced last week. Will ship a fix today and cur new bugfix releases likely tomorrow.

**Reply by Dentrax** [reply]
Same issue on `16.4.9`, but I able to spin up without any issues with version `16.2.2`.

**Reply by hugoShaka** [reply, marked as answer]
16.4.10 and 17.0.4 got released yesterday and should fix the joining issue

**Reply by Dentrax** [reply]
Thank you for such a prompt action. I just tested and can confirmed that both works well!

**Comment by tosih** [comment]
Thanks for this. Worked for us too.
